lib/context.js

```diff
@@ -16,7 +17,13 @@ class ContextSession {
   constructor(ctx, opts) {
     this.ctx = ctx;
     this.opts = Object.assign({}, opts);
-    this.store = this.opts.store;
+    this.store = this.opts.ContextStore ? new this.opts.ContextStore(ctx) : this.opts.store;
+    if (this.store) {
+      assert(typeof this.store.get === 'function', 'store.get must be function');
+      assert(typeof this.store.set === 'function', 'store.set must be function');
+      assert(typeof this.store.destroy === 'function', 'store.destroy must be function');
```

Comment:
No need to check store type on every request.
Only assert once at the beginning.